src/test/assembly/nvptx-kernel-abi/nvptx-kernel-args-abi-v7.rs

```diff
@@ -0,0 +1,258 @@
+// assembly-output: ptx-linker
```

Comment:
Does this really need a post-link output over a plain `assembly`? If it was possible to write this down as a regular assembly test and also to make it `#[no_core]`, this test could be made to run on any machine, and not just when the whole build system is targetting `nvptx64-*` targets.

Reply:
I assume what you mean with "regular assembly test" is to use `// assembly-output: emit-asm` and skip the `// only-nvptx64` tag?

I don't think it will be possible to remove the `// only-nvptx64` tag since the `ptx-kernel` abi is not available for any targets other than `nvptx` targets. Attempting to do it gives me error E0570.

It might be possible to test on llvm-ir, but then we would have to trust LLVM to do the right thing. I'm not sure there is any benefit to this. What do you think?

Reply:
I'm back working on it after Easter. I think this should work fine, but when changing to `//assembly-output: emit-asmi` i hit the following assertion: 
`rustc: /<path>/rust/src/llvm-project/llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp:10352: void llvm::SelectionDAGISel::LowerArguments(const llvm::Function&): Assertion `InVals.size() == Ins.size() && "LowerFormalArguments didn't emit the correct number of values!"' failed.`
Are there any reasons why this assert is only hit when generating assembly as opposed to ptx?

(It seems to be the array tests that hits the assertion)

Reply:
Aha, that seems like a genuine bug in the (LLVM) backend! If I had to guess, I'd say the output from LLVM is wrong in either case already, but the checks that trigger this assertion never get executed when generating ptx directly. I think it is fine to leave the test as-is for now in that case, but definitely should file a follow up issue so that this does not get lost.